rustls/src/client/tls12.rs

```diff
@@ -1063,19 +1062,21 @@ impl<C: CryptoProvider> State<ClientConnectionData> for ExpectFinished<C> {
             _cert_verified: st.cert_verified,
             _sig_verified: st.sig_verified,
             _fin_verified,
+            backend: PhantomData,
         }))
     }
 }
 
 // -- Traffic transit state --
-struct ExpectTraffic {
-    secrets: ConnectionSecrets,
+struct ExpectTraffic<C: CryptoProvider> {
+    secrets: ConnectionSecrets<C>,
     _cert_verified: verify::ServerCertVerified,
     _sig_verified: verify::HandshakeSignatureValid,
     _fin_verified: verify::FinishedMessageVerified,
+    backend: PhantomData<C>,
```

Comment:
The `ConnectionSecrets` type is newly generic since it relies on `C::KeyExchange`. Since it didn't directly use `C` I had to add a PhantomData<C> here and in the `ConnectionSecrets` struct. This feels a little odd (also the field should probably be named `_backend`?).

Maybe the solution should be to make some functions of `ConnectionSecrets` generic, but not the struct? :thinking: